paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/AddPaymentMethodRequirement.kt

```diff
@@ -59,13 +60,25 @@ internal enum class AddPaymentMethodRequirement {
     /** Requires that Instant Debits are possible for this transaction. */
     InstantDebits {
         override fun isMetBy(metadata: PaymentMethodMetadata): Boolean {
-            val paymentMethodTypes = metadata.stripeIntent.paymentMethodTypes
-            val noUsBankAccount = USBankAccount.code !in paymentMethodTypes
-            val supportsBankAccounts = "bank_account" in metadata.stripeIntent.linkFundingSources
-            val isDeferred = metadata.stripeIntent.clientSecret == null
-            return noUsBankAccount && supportsBankAccounts && !isDeferred
+            return metadata.linkMode != LinkMode.LinkCardBrand && metadata.supportsMobileInstantDebitsFlow
+        }
+    },
+
+    /** Requires that LinkCardBrand is possible for this transaction. */
+    LinkCardBrand {
```

Comment:
Do we need to make sure they also have card as an available payment method?

Reply:
No, we only receive the `LinkCardBrand` mode if there’s a `card` type (see [here](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/consumer/elements/helpers/link_passthrough.rb#L457)).